hub/api/tensor.py

```diff
@@ -74,4 +74,7 @@ def numpy(self):
         Returns:
             A numpy array containing the data represented by this tensor.
         """
-        return read_array(self.key, self.provider, self.slice)
+        return read_array(self.key, self.storage, self.slice)
+
+    def flush(self):
```

Comment:
Is this redundant? I feel like this should only ever be called on the Dataset.

Reply:
Yup ideally it should be. We can remove this from here.